ciba-grant-3.2.0.md (#3615)
OAI/OpenAPI-Specification: versions/3.2.0.md

<a name="oauthFlowDeviceAuthorizationUrl"></s>deviceAuthorizationUrl | `string` | `oauth2` (`"deviceAuthorization"`) | **REQUIRED**. The device authorization URL to be used for this flow. This MUST be in the form of a URL. The OAuth2 standard requires the use of TLS.
<a name="oauthFlowTokenUrl"></a>tokenUrl | `string` | `oauth2` (`"password"`, `"clientCredentials"`, `"authorizationCode"`, `"deviceAuthorization"`, `"ciba"`) | **REQUIRED**. The token URL to be used for this flow. This MUST be in the form of a URL. The OAuth2 standard requires the use of TLS.
<a name="oauthFlowRefreshUrl"></a>refreshUrl | `string` | `oauth2` | The URL to be used for obtaining refresh tokens. This MUST be in the form of a URL. The OAuth2 standard requires the use of TLS.
<a name="oauthFlowScopes"></a>scopes | Map[`string`, `string`] | `oauth2` | **REQUIRED**. The available scopes for the OAuth2 security scheme. A map between the scope name and a short description for it. The map MAY be empty.

> <a name="backchannelTokenDeliveryModesSupported"></a>cibaDeliveryModes | Array[`string`] | `oauth2` (`"ciba"`) | **REQUIRED**. JSON array containing one or more of the following values: poll, ping, and push.

Reviewer: ```suggestion
<a name="backchannelTokenDeliveryModesSupported"></a>cibaDeliveryModes | Array[`string`] | `oauth2` (`"ciba"`) | **REQUIRED**. JSON array containing one or more of the following values: `poll`, `ping`, and `push`.
```

> <a name="backchannelTokenDeliveryModesSupported"></a>ciba_delivery_modes | Array[`string`] | `oauth2` (`"ciba"`) | **REQUIRED**. JSON array containing one or more of the following values: poll, ping, and push.

Author: ```suggestion
<a name="backchannelTokenDeliveryModesSupported"></a>cibaDeliveryModes | Array[`string`] | `oauth2` (`"ciba"`) | **REQUIRED**. JSON array containing one or more of the following values: poll, ping, and push.
```

> <a name="backchannel_token_delivery_modes_supported"></a>ciba_delivery_modes | Array[`string`] | `oauth2` (`"ciba"`) | **REQUIRED**. JSON array containing one or more of the following values: poll, ping, and push.

Reviewer 2: OpenAPI uses camelCasing for parameter values.  So although the CIBA specification uses snake case, for consistency we should require tooling to do the translation and maintain consistency in the OpenAPI description.

> <a name="backchannel_token_delivery_modes_supported"></a>ciba_delivery_modes | Array[`string`] | `oauth2` (`"ciba"`) | **REQUIRED**. JSON array containing one or more of the following values: poll, ping, and push.

Author: ```suggestion
<a name="backchannelTokenDeliveryModesSupported"></a>ciba_delivery_modes | Array[`string`] | `oauth2` (`"ciba"`) | **REQUIRED**. JSON array containing one or more of the following values: poll, ping, and push.
```

> <a name="backchannelAuthenticationRequestSigningAlgValues"></a>ciba_signing_algs | Array[`string`] | `oauth2` (`"ciba"`) |  JSON array containing a list of the JWS signing algorithms (alg values) supported by the OP (OpenIdConnect Provider) for signed authentication requests

Author: ```suggestion
<a name="backchannelAuthenticationRequestSigningAlgValues"></a>cibaSigningAlgs | Array[`string`] | `oauth2` (`"ciba"`) |  JSON array containing a list of the JWS signing algorithms (alg values) supported by the OP (OpenIdConnect Provider) for signed authentication requests
```

> <a name="backchannel_authentication_request_signing_alg_values"></a>ciba_signing_algs | Array[`string`] | `oauth2` (`"ciba"`) |  JSON array containing a list of the JWS signing algorithms (alg values) supported by the OP (OpenIdConnect Provider) for signed authentication requests

Author: ```suggestion
<a name="backchannelAuthenticationRequestSigningAlgValues"></a>ciba_signing_algs | Array[`string`] | `oauth2` (`"ciba"`) |  JSON array containing a list of the JWS signing algorithms (alg values) supported by the OP (OpenIdConnect Provider) for signed authentication requests
```

> <a name="backchannel_authentication_request_signing_alg_values"></a>ciba_signing_algs | Array[`string`] | `oauth2` (`"ciba"`) | **OPTIONAL**. JSON array containing a list of the JWS signing algorithms (alg values) supported by the OP (OpenIdConnect Provider) for signed authentication requests

Reviewer 2: ```suggestion
<a name="backchannel_authentication_request_signing_alg_values"></a>ciba_signing_algs | Array[`string`] | `oauth2` (`"ciba"`) |  JSON array containing a list of the JWS signing algorithms (alg values) supported by the OP (OpenIdConnect Provider) for signed authentication requests
```

> <a name="backchannelUserCodeParameterSupported"></a>ciba_user_code | `boolean` | `oauth2` (`"ciba"`) |  Boolean value specifying whether the OP supports the use of the user_code parameter, with true indicating support. If omitted, the default value is false.

Author: ```suggestion
<a name="backchannelUserCodeParameterSupported"></a>cibaUserCode | `boolean` | `oauth2` (`"ciba"`) |  Boolean value specifying whether the OP supports the use of the user_code parameter, with true indicating support. If omitted, the default value is false.
```

> <a name="backchannel_user_code_parameter_supported"></a>ciba_user_code | `boolean` | `oauth2` (`"ciba"`) |  Boolean value specifying whether the OP supports the use of the user_code parameter, with true indicating support. If omitted, the default value is false.

Author: ```suggestion
<a name="backchannelUserCodeParameterSupported"></a>ciba_user_code | `boolean` | `oauth2` (`"ciba"`) |  Boolean value specifying whether the OP supports the use of the user_code parameter, with true indicating support. If omitted, the default value is false.
```

> <a name="backchannel_user_code_parameter_supported"></a>ciba_user_code | `boolean` | `oauth2` (`"ciba"`) | **OPTIONAL**. Boolean value specifying whether the OP supports the use of the user_code parameter, with true indicating support. If omitted, the default value is false.

Reviewer 2: ```suggestion
<a name="backchannel_user_code_parameter_supported"></a>ciba_user_code | `boolean` | `oauth2` (`"ciba"`) |  Boolean value specifying whether the OP supports the use of the user_code parameter, with true indicating support. If omitted, the default value is false.
```

        "write:pets": "modify pets in your account",
        "read:pets": "read your pets"
      }
    },
    "ciba": {
      "authorizationUrl": "https://example.com/api/oauth/dialog",
      "tokenUrl": "https://example.com/api/oauth/token",
      "scopes": {
        "write:pets": "modify pets in your account",
        "read:pets": "read your pets"
      },

>       "ciba_delivery_modes": ["poll"]

Author: ```suggestion
      "cibaDeliveryModes": ["poll"]
```
